src/components/fast_clock.c

```diff
@@ -0,0 +1,253 @@
+/*
+ * @file fast_clock.c
+ * @brief Loconet clock implementation
+ *
+ * \copyright Copyright 2017 /Dev. All rights reserved.
+ * \license This project is released under MIT license.
+ *
+ * @author Jan Martijn van der Werf
+ * @see fast_clock.h
+ */
+
+#include "fast_clock.h"
+
+// ----------------------------------------------------------------------------
+// Prototypes
+void fast_clock_handle_update_(FAST_CLOCK_TIME_Type time);
+
+// ----------------------------------------------------------------------------
+// this is the function that any system that wants to react on the clock should
+// implement.
+//
+__attribute__ ((weak, alias("fast_clock_handle_update_")))
+  void fast_clock_handle_update(FAST_CLOCK_TIME_Type time);
+
+// ----------------------------------------------------------------------------
+// set time initially to sunday 00:00:00.0000
+FAST_CLOCK_TIME_Type current_time = {0, 0, 0, 0};
+
+// ----------------------------------------------------------------------------
+typedef struct {
+  bool master;
+  uint8_t id1;
+  uint8_t id2;
+  uint16_t intermessage_delay;
+  uint8_t rate;
+  uint8_t millisecond;
+} FAST_CLOCK_STATUS_Type;
+
+FAST_CLOCK_STATUS_Type fast_clock_status = {0, 0, 0, 0, 1, 0};
+
+uint16_t fast_clock_current_intermessage_delay = 0;
+
+// ----------------------------------------------------------------------------
+// The clock is defined on div8. That means that the clock is triggered
+// every 50ms.
+#define FAST_CLOCK_TIMER_DELAY 50000
+Tc *fast_clock_timer;
+
+// ----------------------------------------------------------------------------
+//
+void fast_clock_init_timer(Tc *timer, uint32_t pm_tmr_mask, uint32_t gclock_tmr_id, uint32_t nvic_irqn)
+{
+  fast_clock_timer = timer;
+
+  // Enable clock for fast clock, without prescaler
+  PM->APBCMASK.reg |= pm_tmr_mask;
+  GCLK->CLKCTRL.reg =
+    GCLK_CLKCTRL_ID(gclock_tmr_id)
+    | GCLK_CLKCTRL_CLKEN
+    | GCLK_CLKCTRL_GEN(0);
+
+  /* CTRLA register:
+   *   PRESCSYNC: 0x02  RESYNC
+   *   RUNSTDBY:        Ignored
+   *   PRESCALER: 0x03  DIV8, each tick is be 1 us
+   *   WAVEGEN:   0x01  MFRQ, zero counter on match
+   *   MODE:      0x00  16 bits timer
+   */
+  fast_clock_timer->COUNT16.CTRLA.reg =
+    TC_CTRLA_PRESCSYNC_RESYNC
+    | TC_CTRLA_PRESCALER_DIV8
+    | TC_CTRLA_WAVEGEN_MFRQ
+    | TC_CTRLA_MODE_COUNT16;
+
+  fast_clock_timer->COUNT16.INTENSET.reg = TC_INTENSET_MC(1);
+  NVIC_EnableIRQ(nvic_irqn);
+
+  fast_clock_timer->COUNT16.COUNT.reg = 0;
+  fast_clock_timer->COUNT16.CC[0].reg = FAST_CLOCK_TIMER_DELAY;
+  fast_clock_timer->COUNT16.CTRLA.reg |= TC_CTRLA_ENABLE;
+}
+
+// ----------------------------------------------------------------------------
+// sets the values for being a master: whether it is a
```

Comment:
This reads like an unfinished comment. Furthermore, I'd like to suggest to convert the function to `fast_clock_set_master` (and subsequent `fast_clock_set_slave` a little further down).